gtda/homology/_utils.py

```diff
@@ -16,12 +16,29 @@ def _pad_diagram(Xd, homology_dimensions, max_n_points, min_values):
             padding = ((0, n_points_to_pad), (0, 0))
             Xd[dim] = np.pad(Xd[dim], padding, 'constant')
             Xd[dim][-n_points_to_pad:, :] = \
-                [min_values[dim], min_values[dim], dim]
-    Xd = np.vstack([Xd[dim] for dim in homology_dimensions])
+                [min_values[dim], min_values[dim]]
+
+    # Add dimension as the third elements of each (b, d) tuple
+    Xd = [np.hstack([Xd[dim], dim * np.ones((Xd[dim].shape[0], 1),
+                                            dtype=Xd[dim].dtype)])
+          for dim in homology_dimensions]
+
+    Xd = np.vstack([Xd[d] for d in range(len(homology_dimensions))])
```

Comment:
This looks equivalent to simply doing:
```
   Xd = np.vstack(Xd)
```
because `Xd` is defined as a list above.